src/coreclr/src/pal/src/thread/thread.cpp

```diff
@@ -1463,13 +1463,17 @@ CorUnix::GetThreadTimesInternal(
 
 #if HAVE_PTHREAD_GETCPUCLOCKID
     if (pthread_getcpuclockid(pTargetThread->GetPThreadSelf(), &cid) != 0)
-#endif
     {
         ASSERT("Unable to get clock from thread\n", hThread);
         SetLastError(ERROR_INTERNAL_ERROR);
         pTargetThread->Unlock(pThread);
         goto SetTimesToZero;
     }
+#elif __sun
+    cid = CLOCK_REALTIME;
```

Comment:
@jclulow pointed out in the channel that it should be `CLOCK_THREAD_CPUTIME_ID`. I will update.

Reply:
Since we needed lwpstatus of target thread, instead of self, implemented it by reading `/proc/<pid>/lwp/<lwpid>/lwpstatus` while sharing common parts of linux.

@krytarowski, would it make sense if NetBSD branch also use `pTargetThread->GetLwpId()` (instead of current thread's lwpid)?

Reply:
I don't see the code.

Reply:
https://github.com/dotnet/runtime/blob/7ef37fd35853ad437671cbc157c6cf0d6afc2bba/src/coreclr/src/pal/src/thread/thread.cpp#L1406 here i was wondering if `pTargetThread->GetLwpId()` could be used instead of `THREADSilentGetCurrentThreadId()`?

Reply:
Can you split this code into more pull requests?

In general we added in NetBSD the needed interface for `pthread_getcpuclockid`.

Reply:
This code is not written by me neither it is touched by the PR. The question was about seemingly wrong ID being used in the existing NetBSD branch. But if you think `THREADSilentGetCurrentThreadId` is right and there `GetLwpId` does not make sense, then it is fine.